src/classes/repositories/MongodbEventRepository.ts

```diff
@@ -166,6 +166,19 @@ export abstract class MongodbEventRepository<TEntity extends IEventEntity> exten
     return new this._Entity().setPersistedEvents(document.events)
   }
 
+  /**
+   * Verify if a document exist by a query
+   * If the document exist. it will return true
+   * @param {Object} query with mongo query
+   */
+  protected async _existBy (query: { [key: string]: any }): Promise<Boolean> {
```

Comment:
According to the last comment, this could be a public function, since all the implementations would be something like `async existsBy (query: { [key: string]: any }): Promise<boolean> {}`.

This is due to the fact that **all** the implementations of this function will be the same, so there's no need for a protected function inside the base class since the user will always do the same thing here